mergify_engine/engine/queue_runner.py

```diff
@@ -53,6 +54,37 @@ async def are_checks_pending(
     return False
 
 
+async def have_unexpected_changes(
+    ctxt: context.Context, car: merge_train.TrainCar
+) -> bool:
+    expected_commits = (len(car.parent_pull_request_numbers) + 1) * 2
+    have_bad_events = any(
+        (
+            s["event_type"] == "pull_request"
+            and typing.cast(github_types.GitHubEventPullRequest, s["data"])["action"]
+            in ["closed", "reopened"]
+            for s in ctxt.sources
+        )
+    )
```

Comment:
Looks suboptimal to iterate here since it might not be used

Reply:
I was for logging purpose, I can rewrite it